pymeasure/instruments/lecroy/lecroyT3DSO1204.py

```diff
@@ -0,0 +1,1151 @@
+            transmitted_points = int(message_header[-9:])
+            received_points = len(message) - self._header_size - self._footer_size
+            if transmitted_points != received_points:
+                raise ValueError(f"Number of transmitted points ({transmitted_points}) != "
+                                 f"number of received points ({received_points})")
+
+        def _footer_sanity_checks(message):
+            # The footer is always a double line-carriage \n\n
+            message_footer = bytes(message[-self._footer_size:]).decode("ascii")
+            if message_footer != "\n\n":
+                raise ValueError(f"Waveform data in invalid : footer is {message_footer}")
+
+        # First of all set the correct acquisition type
+        if averaging > 1:
+            self.acquisition_type = "average"
+            self.acquisition_average = averaging
+        else:
+            self.acquisition_type = "normal"
+
+        # Check that we are trying to read a reasonable amount of points
+        sample_points = self.acquisition_sample_size(self.waveform_source)
+        if requested_points > 0 and 0 < sample_points < requested_points:
+            raise ValueError(f"Number of requested points ({requested_points}) is greater than "
+                             f"number of sampled points ({sample_points})")
+
+        # Setup waveform acquisition parameters
+        source = _sanitize_source(source)
+        self.waveform_source = source
+        self.waveform_sparsing = sparsing
+        self.waveform_points = requested_points
+        self.waveform_first_point = 0
+
+        # Read waveform preamble (waveform metadata)
+        preamble = self.waveform_preamble
+
+        # Check how many points are to be expected
+        self.write(f"{source}:WF? DAT2")
+        values = self._digitize(source=source)
+        expected_points = _header_sanity_checks(values)
+        if requested_points <= 0:
+            expected_points /= sparsing
+
+        # If the number of points is big enough, split the data in small chunks and read it one
+        # chunk at a time. For less than 500K points we do not bother splitting them.
+        chunk_bytes = 5e5
+        chunk_points = chunk_bytes - self._header_size - self._footer_size
+        iterations = -(expected_points // -chunk_points)
+        i = 0
+        data = []
+        while True:
+            read_points = i * chunk_points
+            remaining_points = expected_points - read_points
+            if remaining_points > chunk_points:
+                requested_points = chunk_points
+            else:
+                requested_points = remaining_points
+            requested_bytes = requested_points + self._header_size + self._footer_size
+            self.waveform_first_point = read_points * sparsing
+            self.waveform_points = requested_points
+            values = self._digitize(source=source, requested_bytes=requested_bytes)
```

Comment:
Do you write the request for every new chunk?

Reply:
Yes, every chunk must be requested individually otherwise the scope always returns the first chunk.